### PR TITLE
Rename `KernelResult` to just `Result`.

### PR DESCRIPTION
**Pull request by wedsonaf**
This is just a rename followed by `rustfmt`, no functional change is
intended.

Signed-off-by: Wedson Almeida Filho <wedsonaf@google.com>

### CODE REVIEW

**Comment by wedsonaf** [comment]
@ojeda this looks to be related to the CI. Is there anything I can do to make this go away?

```
E: The repository 'https://packages.microsoft.com/repos/azure-cli focal Release' no longer has a Release file.
```

**Comment by ojeda** [comment]
It happens from time to time :(

We will just wait a few hours, I guess.

File: rust/kernel/io_buffer.rs [hunk 1]
```diff
@@ -93,10 +93,10 @@ pub trait IoBufferWriter {
     /// # Safety
     ///
     /// The input buffer must be valid.
-    unsafe fn write_raw(&mut self, data: *const u8, len: usize) -> KernelResult;
+    unsafe fn write_raw(&mut self, data: *const u8, len: usize) -> Result;
 
     /// Writes the contents of the given data into the io buffer.
-    fn write<T: WritableToBytes>(&mut self, data: &T) -> KernelResult<()> {
+    fn write<T: WritableToBytes>(&mut self, data: &T) -> Result<()> {
```
Comment (kloenk) [inline rust/kernel/io_buffer.rs]: We could clean all this up, while we are at it.

```suggestion
    fn write<T: WritableToBytes>(&mut self, data: &T) -> Result {
```

Comment (wedsonaf) [reply rust/kernel/io_buffer.rs]: Indeed. Fixed.

**Comment by TheSven73** [comment]
I did a quick read through, and LGTM.

**Review by alex — APPROVED** [review]

**Comment by alex** [comment]
This is going to cause merge conflicts everywhere. Sorry :-(

**Pull request merged**